### PR TITLE
feat(split-button): Add support for appearance

### PR DESCRIPTION
**Pull request by crowjonah**
**Related Issue:** https://github.com/Esri/calcite-components/issues/796

## Summary

Adds a passthrough `appearance` attribute to the `calcite-split-button` that lands on both the primary and secondary button elements, as well as a new `split-child` attribute on the `calcite-button` component to allow for styling specific to when they're inside split buttons!

image

### CODE REVIEW

**Comment by bstifle** [comment]
@crowjonah This is awesome! 

Just one edit that applies to outline and clear:

image


it looks like there's an artifact that appears on hover. its a 2px ui-blue-1 line that should be removed on hover/focus.



Also, would you mind adding these same examples to the demo page for dark theme?

File: src/components/calcite-button/calcite-button.tsx [hunk 1]
```diff
@@ -194,6 +197,7 @@ export class CalciteButton {
       "icon-start",
       "icon-end",
       "id",
+      "isSplitPrimary",
```
Comment (macandcheese) [inline src/components/calcite-button/calcite-button.tsx:200]: This should match `splitChild` above yeah?

**Review by macandcheese — APPROVED** [review]
One cleanup comment - looks good! Thanks for adding this in.

File: src/components/calcite-button/calcite-button.tsx (hunk 1 above)
Comment (crowjonah) [reply src/components/calcite-button/calcite-button.tsx:200]: Totally! Thanks for catching

**Comment by crowjonah** [comment]
> Also, would you mind adding these same examples to the demo page for dark theme?

Thanks for taking a look! @bstifle, how are these? I think I got that separator hover color sorted out across all possible permutations, but I could've missed something!
image

**Comment by bstifle** [comment]
@crowjonah nice! those fixes look good.

Here are some for dark theme:

1. transparent, (Light Color) dark theme hover state should be using 255,255,255 for the color instead of 0,0,0:
image


2. outline appearance (Light Color) dark theme hover states use a gray border, should be the same #151515 border, just thicker
image

File: src/components/calcite-split-button/calcite-split-button.scss [hunk 2]
```diff
@@ -39,6 +77,64 @@
...
+      }
+      .split-button__divider-container {
+        background-color: transparent;
+        border-color: var(--calcite-button-dark);
+      }
+    }
+  }
+  &:host([theme="dark"]) {
+    .split-button__divider-container {
+      background-color: var(--calcite-button-dark);
```
Comment (macandcheese) [inline src/components/calcite-split-button/calcite-split-button.scss:122]: The CSS vars in the top of this file could be replaced with theme-aliased CSS vars from calcite-colors, which could cut down on some of these dark specific rules: https://github.com/Esri/calcite-colors/blob/731942192f939993ceade97cffe2649724b56fd5/dist/colors.scss#L506

**Comment by bstifle** [comment]
nice work @crowjonah ship it! 📦

**Comment by crowjonah** [comment]
> The CSS vars in the top of this file could be replaced with theme-aliased CSS vars from calcite-colors, which could cut down on some of these dark specific rules: https://github.com/Esri/calcite-colors/blob/731942192f939993ceade97cffe2649724b56fd5/dist/colors.scss#L506

I reworked the whole style file to reflect what's happening these days in `calcite-button`, eliminating property overrides, and instead relying wholly on the appearance/theme/color variable combinations. @macandcheese, the diff isn't gonna be super helpful, but does this look better? @bstifle, probably worth another glance to make sure I didn't mess it up again :)

**Comment by macandcheese** [comment]
Rad, thanks @crowjonah - @bstifle can you sign off on this and we can ⛵  it?

**Comment by crowjonah** [comment]
Got @bstifle's ok offline! There's some wonkiness with the transparent and clear light/dark color/theme combos inherent to the `calcite-button` children that's logged here: https://github.com/Esri/calcite-components/issues/576.

**Pull request merged**